runtime/flang/close.c

```diff
@@ -65,15 +65,14 @@ __fortio_close(FIO_FCB *f, int flag)
     else if (f->status == FIO_SCRATCH)
       unlink(f->name);
 #endif
-
-    free(f->name);
   } else { /* stdin, stdout, stderr - just flush */
 #if defined(TARGET_OSX)
     if (f->unit != 5 && f->unit != -5)
 #endif
       if (__io_fflush(f->fp) != 0)
         return __fortio_error(__io_errno());
   }
+  free(f->name);
 
   __fortio_free_fcb(f); /*  free up FCB for later use  */
```

Comment:
but why __fortio_free_fcb() does not take care of it?

Reply:
That's a very good point. I'll change the code.

Reply:
Note that this function has lots of early `return`s in the middle. Is it safe to assume each early return means the FCB is still in use and should not be freed and nothing is leaking in memory?

Reply:
I checked the error handling code. It seems that in all cases, the runtime will print an error messages, and either aborts the process completely, or stops all further read/write operations (`fioFcbTbls.error = TRUE`). Not freeing the FCB probably isn't a problem.